subprojects/core-api/src/main/java/org/gradle/api/tasks/Nested.java

```diff
@@ -30,7 +30,12 @@
  * <p>The implementation of the nested bean is tracked as an input, too.
  * This allows tracking behavior such as {@link org.gradle.api.Action}s as task inputs.</p>
  *
- * <p>This annotation supports iterable values by treating each element as a separate nested input.</p>
+ * <p>This annotation supports iterable values by treating each element as a separate nested bean.
+ * If an element of the iterable implements {@link org.gradle.api.Named}, then the name is used as property name.
+ * If it does not, the index in the iterable prefixed by {@code $} is used, e.g. {@code $0}</p>
```

Comment:
Looks like a good place to mention that consistent ordering of the iterable is crucial for reliable up-to-date checks and caching. WDYT?

Reply:
We should probably also mention that multiple entries with the same name are not allowed.

Reply:
Done.